File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/MobileConfigurations/Loan/Calculators/LoanCalculator.cs
namespace Psi.Data.Models.ClientConfigurationModels.MobileConfigurations.Loan.Calculators
{
	public class LoanCalculator : SettingsBaseHelper
	{
		private readonly ISettingsBase _settingsBase;
		private AutoLoan _autoLoan;

		public LoanCalculator(ISettingsBase settingsBase) : base(settingsBase)
		{
			_settingsBase = settingsBase;
		}

		public AutoLoan AutoLoan
		{
			get { return _autoLoan ?? (_autoLoan = new AutoLoan(_settingsBase)); }
			set { _autoLoan = value; }
		}
	}
}

File: infrastructure/models/ClientConfigurationModels/MobileConfigurations/Loan/Calculators/LoanCalculator.ts
import { Setting, ISettingsGroup, ISettingsMetadata } from '@models/base/types';
import { AutoLoan } from '../AutoLoan';
export interface LoanCalculatorConfig {
    AutoLoan: AutoLoan;
}

export class LoanCalculator implements ISettingsGroup {
    static readonly metadata: ISettingsMetadata = {
        groupName: 'LoanCalculator'
    };


            private _autoLoan: AutoLoan;
            get autoLoan(): AutoLoan {
                return this._autoLoan;
            }
            set autoLoan(value: AutoLoan) {
                this._autoLoan = value;
            }


    constructor() {}


        toSettings(): Setting[] {
            return [
                { key: "LoanCalculator.AutoLoan", value: this._autoLoan, dataType: 'autoloan', label: "Auto Loan" },
            ];
        }

}